src/main/java/ua/danit/final_project/controllers/security/AuthenticationController.java

```diff
@@ -0,0 +1,101 @@
+package ua.danit.final_project.controllers.security;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.DisabledException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import ua.danit.final_project.configuration.security.JwtAuthenticationRequest;
+import ua.danit.final_project.configuration.security.JwtAuthenticationResponse;
+import ua.danit.final_project.configuration.security.JwtTokenUtil;
+import ua.danit.final_project.controllers.exceptions.AuthenticationException;
+import ua.danit.final_project.dto.DefaultMapper;
+import ua.danit.final_project.dto.UserDto;
+import ua.danit.final_project.entities.User;
+import ua.danit.final_project.services.RegistrationService;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Objects;
+
+@RestController
+public class AuthenticationController {
+
+  @Value("${jwt.header}")
+  private String tokenHeader;
+
+  private final AuthenticationManager authenticationManager;
+  private final JwtTokenUtil jwtTokenUtil;
+  private final UserDetailsService userDetailsService;
+  private final RegistrationService registrationService;
+  private final DefaultMapper mapper;
+
+  public AuthenticationController(AuthenticationManager authenticationManager,
+                                  JwtTokenUtil jwtTokenUtil,
+                                  @Qualifier("jwtUserDetailsService") UserDetailsService userDetailsService,
+                                  RegistrationService registrationService,
+                                  DefaultMapper mapper) {
+    this.authenticationManager = authenticationManager;
+    this.jwtTokenUtil = jwtTokenUtil;
+    this.userDetailsService = userDetailsService;
+    this.registrationService = registrationService;
+    this.mapper = mapper;
+  }
+
+  @PostMapping("/auth")
+  public ResponseEntity<?> createAuthenticationToken(@RequestBody JwtAuthenticationRequest authenticationRequest)
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove usage of generic wildcard type. ([squid:S1452](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AS1452))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=173&issues=AWb1IgCVW3Y69bz5KWFz&open=AWb1IgCVW3Y69bz5KWFz)

[ISSUE_KEY]: AWb1IgCVW3Y69bz5KWFz